connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -2461,48 +2475,39 @@ class SparkConnectPlanner(val sessionHolder: SessionHolder) extends Logging {
 
   def process(
       command: proto.Command,
-      responseObserver: StreamObserver[ExecutePlanResponse],
-      executeHolder: ExecuteHolder): Unit = {
+      responseObserver: StreamObserver[ExecutePlanResponse]): Unit = {
```

Comment:
Conceptually, this shouldn't be an issue because it's not really a public API. but it's not backwards compatible.

This is particularly interesting because you went through some length to fix this problem in the constructor though.

@hvanhovell Do we consider the SparkConenct planner to be public? Probably, not correct?

Reply:
FWIW, between Spark 3.4 and Spark 3.5, these interfaces also changed.
For example https://github.com/apache/spark/pull/41618 made it take a SessionHolder instead of SparkSession, so that various parameters don't have to be passed to it lossely. This PR adding ExecuteHolder to it is in a very similar spirit.

Reply:
No, I don't think we should. In the end that will hamper our ability to evolve the interface.